### ISSUE TITLE
Issue for visualization

### ISSUE THREAD

**Issue by leihouyeung** [uncbiag/robot#1]
Thanks for such an excellent work. When I tried to visualize the point cloud, the following issue was raised:
```
[xcb] Unknown request in queue while appending request
[xcb] Most likely this is a multi-threaded client and XInitThreads has not been called
[xcb] Aborting, sorry about that.
python3: ../../src/xcb_io.c:151: append_pending_request: Assertion `!xcb_xlib_unknown_req_pending' failed.
```
It looks like the issue happened on the 'p.write_frame()' function.

Hope for your answer. Thanks。

**Comment by marcniethammer** [comment]
Sounds like it might be an issue with the xserver. Are you running this remotely?

**Comment by leihouyeung** [comment]
Yes, I ran it remotely. Is there any way I could save the GIF?

**Comment by marcniethammer** [comment]
Seems like a problem with pyvista and the xserver. @hbgtjxzbbx do you know what could be causing this?

**Comment by leihouyeung** [comment]
I tried to run locally. But it did not work either.

**Comment by marcniethammer** [comment]
With the same error message?

**Comment by leihouyeung** [comment]
Yes. I tried to fix the bug but in vain.  I have to use another way of visualization now.  Thanks for your response.

**Comment by hbgtjxzbbx** [comment]
Sorry for the late reply. if you run it remotely in headless display mode, please follow pyvista installation guide.
https://docs.pyvista.org/getting-started/installation.html

sudo apt-get install libgl1-mesa-dev
sudo apt-get install lxvfb

Then, you need to configure the headless display, for MyBinder, create a file called start and include the following set up script that will run every time your Docker container is launched:

#!/bin/bash
set -x
export DISPLAY=:99.0
export PYVISTA_OFF_SCREEN=true
export PYVISTA_USE_IPYVTK=true
which Xvfb
Xvfb :99 -screen 0 1024x768x24 > /dev/null 2>&1 &
sleep 3
set +x
exec "$@"

**Comment by threeneedone** [comment]
colab
